### PR TITLE
chore(build): improve docker-build-push workflow

### PR DESCRIPTION
**Pull request by JohannesRudolph**
- use GITHUB_TOKEN instead of a personal access token
  this is now supported by GitHub and recommended over PATs
- enforce lowercase for repo name so that private forks can more easily
  publish their own builds of unipipe-service-broker to ghcr.io registry

### CODE REVIEW

**Comment by JohannesRudolph** [comment]
note: after merging this PR I shall remove the CR_PAT secret from this repo/org

**Comment by pubudug** [comment]
Seems no reviewers on this PR. I am not that familiar with github workflows. Maybe @SimonTheLeg or @henryde can take a look?

**Comment by JohannesRudolph** [comment]
It's not a lot of changes going on here, all I did here was setting up the build to use a different authentication token so that forks can more easily use GH actions to make private builds and publish them. You can see these changes in action in my fork's workflow run  https://github.com/JohannesRudolph/unipipe-service-broker/actions/runs/814628427 and docker container https://github.com/users/JohannesRudolph/packages/container/package/unipipe-service-broker

**Review by SimonTheLeg — APPROVED** [review]

**Comment by SimonTheLeg** [comment]
> Seems no reviewers on this PR

I will be looking into this. With the latest change GH should automatically add you as reviewers, because you are code-owners 🤔

**Comment by JohannesRudolph** [comment]
@SimonTheLeg any idea why kodiak failed to merge this one? I only see `⚠️ problem contacting GitHub API with method 'merge'` in status.

**Comment by SimonTheLeg** [comment]
it has its hickups from time and time. I've re-added the label

**Comment by JohannesRudolph** [comment]
hm, failed the second time... I've rebased against master manually

**Comment by JohannesRudolph** [comment]
> note: after merging this PR I shall remove the CR_PAT secret from this repo/org

done

**Pull request merged**
